### PR TITLE
Return Error::TangleInclusion instead of Error::Node(NotFound(_))

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Return Error::TangleInclusion instead of Error::Node(NotFound(_))
Add conflicting tx test

## Links to any relevant issues

#387 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running the test

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/src/wallet/account/operations/retry.rs [hunk 1]
```diff
@@ -119,8 +119,9 @@ where
                 // After we checked all our reattached blocks, check if the transaction got reattached in another block
                 // and confirmed
                 if conflicting {
-                    let included_block = self.client().get_included_block(transaction_id).await?;
-                    return Ok(included_block.id());
+                    if let Ok(included_block) = self.client().get_included_block(transaction_id).await {
```
Comment (thibault-martinez) [inline sdk/src/wallet/account/operations/retry.rs]: Is there a specific error that should end up in `Error::TangleInclusion`? Aren't we ignoring a lot of potentially unrelated error that maybe should be propagated?

Comment (Thoralf-M) [reply sdk/src/wallet/account/operations/retry.rs]: Don't think there can be a lot of potentially unrelated errors, but changed it

**Review by thibault-martinez — DISMISSED** [review]

File: sdk/src/wallet/account/operations/retry.rs [hunk 2]
```diff
@@ -119,8 +119,14 @@ where
                 // After we checked all our reattached blocks, check if the transaction got reattached in another block
                 // and confirmed
                 if conflicting {
-                    let included_block = self.client().get_included_block(transaction_id).await?;
-                    return Ok(included_block.id());
+                    match self.client().get_included_block(transaction_id).await {
+                        Ok(included_block) => return Ok(included_block.id()),
+                        // If no block was found with this transaction id, then it can't get included
+                        Err(crate::client::Error::Node(crate::client::node_api::error::Error::NotFound(_))) => {
```
Comment (DaughterOfMars) [inline sdk/src/wallet/account/operations/retry.rs]: maybe just do a map_err instead?

```rust
let included_block = 
    self.client()
        .get_included_block(transaction_id)
        .await
        .map_err(|e| if matches!(e, crate::client::Error::Node(crate::client::node_api::error::Error::NotFound(_))) {
            crate::client::Error::TangleInclusion(block_id.to_string())
        } else {
            e
        })?;
```

Maybe there's even a cleaner way

File: sdk/src/wallet/account/operations/retry.rs [hunk 3]
```diff
@@ -119,7 +119,17 @@ where
                 // After we checked all our reattached blocks, check if the transaction got reattached in another block
                 // and confirmed
                 if conflicting {
-                    let included_block = self.client().get_included_block(transaction_id).await?;
+                    let included_block = self.client().get_included_block(transaction_id).await.map_err(|e| {
+                        if matches!(
+                            e,
+                            crate::client::Error::Node(crate::client::node_api::error::Error::NotFound(_))
```
Comment (DaughterOfMars) [inline sdk/src/wallet/account/operations/retry.rs]: You could import at least `crate::client::Error` to make this shorter

Comment (Thoralf-M) [reply sdk/src/wallet/account/operations/retry.rs]: Happy with the change?

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

File: sdk/src/wallet/account/operations/retry.rs (hunk 3 above)
Comment (DaughterOfMars) [reply sdk/src/wallet/account/operations/retry.rs]: Sort of. Still feels like it could be cleaner somehow, but it's fine 🤷‍♀️

**Pull request merged**